### ISSUE TITLE
scattergl lines don't link horizontal segments

### ISSUE THREAD

**Issue by wangyueming** [plotly/plotly.js#2397]
Version: Both 1.33.1 and 1.34.0.

I am sorry I can not reproduce it in codepen. I can reproduce it in Electron of version 1.7.x. Need I prepare an Electron sample?

About the code, you can refer to https://codepen.io/yueming0799/pen/mXLwYx?editors=1010, yet it works fine in codepen.

![2018-02-23 10 56 39](https://user-images.githubusercontent.com/4221783/36576035-21cd7218-1889-11e8-83b9-b11c0bd05318.png)

If the mode is lines, the line is broken.

```

traces.push({
    x: x,
    y: y,
    mode: "lines",
    line: {
      color: "#00ff40",
      width: 2,
      dash: 'solid',
    },
    type:'scattergl'
  });

```

**Comment by etpinard** [comment]
> Need I prepare an Electron sample?

Yes, that would nice. Using a https://gist.github.com/ would be best.

At the very least, could you describe what's _broken_ about it? What do you see? Any console errors? etc?

**Comment by alexcjohnson** [comment]
The data in the codepen is different from what you've got in your screenshot. You can use `Plotly.Plots.graphJson(graphDiv)` to pull out the exact plot and put that into codepen, see if that shows what you want it to.

I'm assuming what you mean by "broken" is that there are gaps in the line - which implies that some of the data was interpreted as non-numeric. Does the same thing happen if you switch to regular `scatter`, not `scattergl`?

**Comment by wangyueming** [comment]
@alexcjohnson yes, I mean there are gaps in the line. And it works as regular scatter.

I made a work around to fix it.

```
  for (let i = 0, l = data.length; i < l; i++) {
    const d = data[i];
    if (d.mode === 'lines') {
      d.mode = 'lines+markers';
      d.marker = {
        color: d.line.color,
        size: d.line.width,
        symbol: 'circle'
      };
    }
  }
  Plotly.newPlot(Setting.GRAPH_DIV, data, layout);

```

**Comment by wangyueming** [comment]
@etpinard I uploaded a project in git@github.com:wangyueming/plotly_bug_repeat.git.
Git clone it and run npm install & npm start, you will see the result.
![4966108e-1791-4da9-9f62-2380b4abd8a3](https://user-images.githubusercontent.com/10303419/36664503-e30d9e92-1b1f-11e8-82ad-6970b4cc4143.png)

**Comment by wangyueming** [comment]
I got it repeated in codepen.
https://codepen.io/yueming0799/pen/mXLwYx?editors=0010
![0d876afa-cb69-48e7-b337-54802d42257d](https://user-images.githubusercontent.com/10303419/36665497-b1022028-1b22-11e8-8031-f0ebc61f3715.png)

**Comment by etpinard** [comment]
Thanks very much @wangyueming 

This is a side-effect from the scattergl regl push. Versions previous to `1.33.0` appear to be working fine.

See https://codepen.io/etpinard/pen/oEaaVq?editors=1010 for a comparison.

cc @dfcreative (this is still a problem on `master`)

**Comment by etpinard** [comment]
Fixed in  #2564
